Merge `Run Cypress OSS tests in isolation` PR from `release-x.37.x` branch (#14372)

nemanjaglumac opened the pull request:
### Status
PENDING CI

### What does this PR accomplish?
- It migrates (merges) a huge commit that dealt with Cypress EE test isolation and improvements (#14372) - we need it in master sooner rather than later to:
1. avoid merge conflicts
2. introduce cleaner test architecture

- Closes #13989

nemanjaglumac committed:
Run Cypress OSS tests in isolation (#14372)

* Run all tests in isolation for `frontend/test/metabase/scenarios/permissions.cy.spec.js`

* Run all tests in isolation for `frontend/test/metabase/scenarios/admin/databases/add.cy.spec.js`

* Run all tests in isolation for `frontend/test/metabase/scenarios/admin/datamodel/hide_tables.cy.spec.js`

* Run all tests in isolation for `frontend/test/metabase/scenarios/admin/datamodel/table.cy.spec.js`

* Run all tests in isolation for `frontend/test/metabase/scenarios/admin/databases/list.cy.spec.js`

* Run all tests in isolation for `frontend/test/metabase/scenarios/question/data_ref.cy.spec.js`

* Run all tests in isolation for `frontend/test/metabase/scenarios/question/operators.cy.spec.js`

* Run all tests in isolation for `frontend/test/metabase/scenarios/question/settings.cy.spec.js`

* Run all tests in isolation for `frontend/test/metabase/scenarios/question/trendline.cy.spec.js`

* Run all tests in isolation for `frontend/test/metabase/scenarios/reference/databases.cy.spec.js`

* Run all tests in isolation for `frontend/test/metabase/scenarios/question/downloads.cy.spec.js`

* Run all tests in isolation for `frontend/test/metabase/scenarios/admin/datamodel/segments.cy.spec.js`

* Run all tests in isolation for `frontend/test/metabase/scenarios/admin/datamodel/metrics.cy.spec.js`

* Run all tests in isolation for `frontend/test/metabase/scenarios/admin/permissions/permissions.cy.spec.js`

* Run all tests in isolation for `frontend/test/metabase/scenarios/admin/settings/spinner.cy.spec.js`

* Run all tests in isolation for `frontend/test/metabase/scenarios/auth/search.cy.spec.js`

* Run all tests in isolation for `frontend/test/metabase/scenarios/auth/signin.cy.spec.js`

* Run all tests in isolation for `frontend/test/metabase/scenarios/dashboard/dashboard_data_permissions.cy.spec.js`

* Run all tests in isolation for `frontend/test/metabase/scenarios/dashboard/dashboard-drill.cy.spec.js`

* Run all tests in isolation for `frontend/test/metabase/scenarios/dashboard/embed.cy.spec.js`

* Run all tests in isolation for `frontend/test/metabase/scenarios/dashboard/title-drill.cy.spec.js`

* Run all tests in isolation for `frontend/test/metabase/scenarios/question/loading.cy.spec.js`

* Run all tests in isolation for `frontend/test/metabase/scenarios/dashboard/nested-cards.cy.spec.js`

* Run all tests in isolation for `frontend/test/metabase/scenarios/dashboard/text-box.cy.spec.js`

* Run all tests in isolation for `frontend/test/metabase/scenarios/home/activity-page.cy.spec.js`

* Run all tests in isolation for `frontend/test/metabase/scenarios/question/native_subquery.cy.spec.js`

* Run all tests in isolation for `frontend/test/metabase/scenarios/visualizations/drillthroughs/chart_drill.cy.spec.js`

* Run all tests in isolation for `frontend/test/metabase/scenarios/question/nested.cy.spec.js`

* Run all tests in isolation for `frontend/test/metabase/scenarios/question/view.cy.spec.js`

* Run all tests in isolation for `frontend/test/metabase/scenarios/visualizations/drillthroughs/dash_drill.cy.spec.js`

* Run all tests in isolation for `frontend/test/metabase/scenarios/home/overworld.cy.spec.js`

* Run all tests in isolation for `frontend/test/metabase-db/postgres/custom-column.cy.spec.js`

* Run all tests in isolation for `frontend/test/metabase/scenarios/alert/alert.cy.spec.js`

* Run all tests in isolation for `frontend/test/metabase/scenarios/dashboard/parameters-embedded.cy.spec.js`

* Run all tests in isolation for `frontend/test/metabase/scenarios/admin/settings/settings.cy.spec.js`

* Run all tests in isolation for `frontend/test/metabase/scenarios/admin/databases/edit.cy.spec.js`

* Run all tests in isolation for `frontend/test/metabase/scenarios/alert/alert.cy.spec.js`

* Run all tests in isolation for `frontend/test/metabase/scenarios/dashboard/permissions.cy.spec.js`

* Run all tests in isolation for `frontend/test/metabase/scenarios/pulse/pulse.cy.spec.js` [ci skip]

* Run all tests in isolation for `frontend/test/metabase/scenarios/question/snippets.cy.spec.js` (#14380)

flamber reviewed (approved):
Reviewed in other PR

nemanjaglumac committed:
Fix `downloads.cy.spec.js`

This file was different on `master`, and that's why the version from `release-x.37.x` branch didn't work.
It had to be adjusted to be able to run in isolation.

nemanjaglumac committed:
Fix `pulse.cy.spec.js`
